### PR TITLE
Updated to use ansible modules for pip, file (rm) and git plus tidy

### PR DESCRIPTION
**Pull request by markaboulton**
As suggested in previous pull request the code has been updated to use the pip, file and git (not suggested but found so added) modules instead of running commands directly.

runtime.yml modified for consistency

### CODE REVIEW

File: ansible/roles/daliuge_runtime/tasks/daliuge.yml [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
+    dest: daliuge
+    version: "{{daliuge_version}}"
 
-- name: Install DALiuGE
-  command: pip install -U .
-  args:
-      chdir: daliuge/
+# Install DALiuGE
+- pip:
+    name: .
```
Comment (oneswig) [inline ansible/roles/daliuge_runtime/tasks/daliuge.yml:21]: Interesting, I've not seen this mode of usage before - does it work?  It looks a little similar to the requirements parameter in http://docs.ansible.com/ansible/latest/pip_module.html#options

File: ansible/roles/daliuge_runtime/tasks/runtime.yml [hunk 2]
```diff
@@ -7,5 +7,5 @@
 - name: Ensure selected packages are installed
   yum:
     name: "{{item}}"
-    state: present
-  with_items: "{{daliuge_packages}}"
+    state: installed
+    with_items: "{{daliuge_packages}}"
```
Comment (oneswig) [inline ansible/roles/daliuge_runtime/tasks/runtime.yml]: The original indentation is actually correct - the with_items iteration is for invocations of yum.

File: ansible/roles/daliuge_runtime/tasks/daliuge.yml [hunk 3]
```diff
@@ -1,17 +1,26 @@
 ---
-- name: Update and/or install pip packages
-  command: pip install -U pip setuptools
+# Update and/or install pip packages
+- pip:
+    name: pip setuptools
```
Comment (oneswig) [inline ansible/roles/daliuge_runtime/tasks/daliuge.yml:4]: Does this syntax work?  I'd have expected either ['pip', 'setuptools'] or its multi-line syntactic equivalent.

**Review by oneswig — COMMENTED** [review]
All looks good, with a few questions on syntax I'm not familiar with.  If you can confirm that it does what you expect we'll merge it in.

File: ansible/roles/daliuge_runtime/tasks/daliuge.yml (hunk 3 above)
Comment (markaboulton) [reply ansible/roles/daliuge_runtime/tasks/daliuge.yml:4]: I believe support for multiples was added recently, I include the output from the run to show this works.

changed: [daliuge-markb-island-mgr-0] => {"changed": true, "cmd": "/bin/pip2 install -U pip setuptools", "failed": false, "name": ["pip setuptools"], "requirements": null, "state": "latest", "stderr": "", "stderr_lines": [], "stdout": "Collecting pip\n  Downloading pip-9.0.1-py2.py3-none-any.whl (1.3MB)\nCollecting setuptools\n  Downloading setuptools-36.6.0-py2.py3-none-any.whl (481kB)\nInstalling collected packages: pip, setuptools\n  Found existing installation: pip 8.1.2\n    Uninstalling pip-8.1.2:\n      Successfully uninstalled pip-8.1.2\n  Found existing installation: setuptools 0.9.8\n    Uninstalling setuptools-0.9.8:\n      Successfully uninstalled setuptools-0.9.8\nSuccessfully installed pip-9.0.1 setuptools-36.6.0\n", "stdout_lines": ["Collecting pip", "  Downloading pip-9.0.1-py2.py3-none-any.whl (1.3MB)", "Collecting setuptools", "  Downloading setuptools-36.6.0-py2.py3-none-any.whl (481kB)", "Installing collected packages: pip, setuptools", "  Found existing installation: pip 8.1.2", "    Uninstalling pip-8.1.2:", "      Successfully uninstalled pip-8.1.2", "  Found existing installation: setuptools 0.9.8", "    Uninstalling setuptools-0.9.8:", "      Successfully uninstalled setuptools-0.9.8", "Successfully installed pip-9.0.1 setuptools-36.6.0"], "version": null, "virtualenv": null}

File: ansible/roles/daliuge_runtime/tasks/daliuge.yml (hunk 1 above)
Comment (markaboulton) [reply ansible/roles/daliuge_runtime/tasks/daliuge.yml:21]: From memory the dot is just saying install the package at the local directory, so looks for a setup.py file to run.

File: ansible/roles/daliuge_runtime/tasks/runtime.yml (hunk 2 above)
Comment (markaboulton) [reply ansible/roles/daliuge_runtime/tasks/runtime.yml]: See update

**Comment by markaboulton** [comment]
Updated code to fix indentation of "with_items" that was accidentally changed.

Rerun this latest update with no errors.

**Review by oneswig — APPROVED** [review]
looks good thanks.

**Pull request merged**